### PR TITLE
move float8_experimental to torchao/float8

### PR DESCRIPTION
**Pull request by vkuzo**
Summary:

This PR moves https://github.com/pytorch-labs/float8_experimental to torchao/float8.

There are no logic changes here. Here is how to reproduce this PR:
* copy float8_experimental/float8_experimental/* to torchao/float8
* copy float8_experimental/test/* to test/float8
* copy float8_experimental/benchmarks/* to benchmarks/float8
* copy the README over and delete sections which no longer apply (license, installation)
* replace `float8_experimental` with `torchao.float8` everywhere
* update tests to skip old PyTorch versions (only nightly supported)
* update non-emulated tests to require an H100 for now (we can enable CI in separate PR)
* update distributed tests to not run in CI for now (we can enable in separate PR)

Test Plan:

```
// run local tests, they pass
./test/float8/test_everything.sh

// run every benchmark in `benchmarks/float8`, they still work
```

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

**Comment by vkuzo** [comment]
> If we wanted to change some of the file naming, we do float8/float8_.. might be a good opportunity, I think since its already namespaced under 'float8' folder it would make sense to remove the leading 'float8_'. I know you will likely not want to complicate the move though, so merely an observation

I'd accept a PR if someone wants to clean that up.  Note: last week we cleaned up the user facing UX and exposed it through the top level `__init__.py` file, so file renames like you suggest are no longer user facing.

**Review by drisspg — APPROVED** [review]

**Comment by jerryzh168** [comment]
thoughts on torchao/float8 v.s. torchao/dtypes/float8?

File: README.md [hunk 1]
```diff
@@ -85,6 +85,12 @@ In some cases we rewrote popular GenAI models to be significantly faster in nati
 
 ### Training
 
+#### Float8
+
+[torchao.float8](torchao/float8) implements training recipes with the scaled float8 dtypes, as laid out in https://arxiv.org/abs/2209.05433.
```
Comment (msaroufim) [inline README.md:90]: wanna mention a topline speedup marketing number?

**Comment by vkuzo** [comment]
> thoughts on torchao/float8 v.s. torchao/dtypes/float8?

Sure, my reasoning was that `float8_experimental` is a workflow, similar to `torchao/quantization` or `torchao/sparsity`.  There is a `Float8Tensor` object inside of `torchao/float8` which is a private API, if in the future that becomes a public API it could make sense to move it into `torchao/dtypes`.

File: README.md (hunk 1 above)
Comment (vkuzo) [reply README.md:90]: our last public number is from 2023H2, we plan to release new speedups in ~weeks but not ready yet.  Will add it here when it's posted.

**Review by msaroufim — APPROVED** [review]

**Comment by awgu** [comment]
what happens to float8_experimental now? is it deprecated?

**Comment by vkuzo** [comment]
> what happens to float8_experimental now? is it deprecated?

I'm finishing up the migration today, just wanted to verify nothing was broken before I go ahead, which I have verified this morning.  float8_experimental will be archived and get a README.md section pointing people here for the new code location.

**Pull request merged**